edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -433,8 +437,50 @@ static Task Cleanup(Option<Agent> agentOption, ILogger logger)
                 logger.LogError(AgentEventIds.Agent, ex, "Error on shutdown");
                 return Task.CompletedTask;
             }
-        }
-
+        }
+
+        static void ValidateStorageIdentity(string storagePath, string deviceId, string iotHubHostname, string moduleId, ILogger logger, string moduleGenerationId = "")
+        {
+            string[] metadata = new string[4];
+            metadata[0] = deviceId;
+            metadata[1] = iotHubHostname;
+            metadata[2] = moduleId;
+            metadata[3] = moduleGenerationId;
+            string file = Directory.GetFiles(storagePath, "DEVICE_IDENTITY")[0];
+            try
+            {
+                int counter = 0;
+                foreach (string line in File.ReadLines(file))
+                {
+                    if (!line.Equals(metadata[counter]))
+                    {
+                        logger.LogInformation("Different Device Identity. Deleting local storage.");

```

Comment:
suggestion: "Persistent storage is for a different device identity {actual}, then the current {current}. Deleting local storage."